### PR TITLE
Add oauth check before get to limit oath requests

### PR DESCRIPTION
**Pull request by Tombo1001**
Add oauth 'check' before 'get' to limit oath requests.
Also fixed hardcoded schedule link - now dynamic using .env twitch ID.

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -293,14 +304,30 @@ def get_oauth(client_id, client_secret, grant_type):
...
-        global oauth_timer
-        oauth_timer = (json_response['expires_in'])
         new_oauth_token = str((json_response['access_token']))
+    else:
+        print("Unable to obtain Twitch API OAUTH token")
+        new_oauth_token = ""
     return new_oauth_token
 
 
+def check_oath(oauth_token, client_id):
```
Comment (biodrone) [inline bot.py:314]: Should this be check_oauth instead of check_oath?

File: bot.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ async def on_message(message):
 
     # context sensitive command - scheudle
     if '!schedule' in message.content:
-        response = 'Check out our stream schedule here: https://www.twitch.tv/jennytree95/schedule'
+        response = 'Check out our stream schedule here: https://www.twitch.tv/' + twitch_userID + '/schedule'
```
Comment (biodrone) [inline bot.py:155]: Can we use a format string here instead? Just looks a bit cleaner.

```suggestion
        response = f'Check out our stream schedule here: https://www.twitch.tv/{twitch_userID}/schedule'
```

**Review by biodrone — CHANGES_REQUESTED** [review]
Couple of tweaks, if they make sense?

File: bot.py (hunk 1 above)
Comment (Tombo1001) [reply bot.py:314]: **oath**
_noun_
a solemn promise, often invoking a divine witness, regarding one's future action or behaviour.
"they took an oath of allegiance to the king"
....refactoring....

File: bot.py (hunk 2 above)
Comment (Tombo1001) [reply bot.py]: Yarp job done, like this style. I will make it my standard.

File: bot.py (hunk 1 above)
Comment (Tombo1001) [reply bot.py]: problem solved my liege.

**Review by biodrone — APPROVED** [review]
Noice 👍

**Pull request merged**